xla/xla_data.proto

```diff
@@ -933,7 +933,18 @@ message PrecisionConfig {
 
     // Next: 4
   }
-  repeated Precision operand_precision = 1;
+  enum Gradient {
+    UNSET = 0;
+    ACTIVATION = 1;
+    GRADIENT = 2;
+  }
+
+  message PrecisionGradient {
+    Precision precision = 1;
+    Gradient gradient = 2;
+  };
+
+  repeated PrecisionGradient operand_precision = 1;
```

Comment:
I'm wondering if it would be a more compatible change to do this:
```
repeated Precision operand_precision = 1;
repeated Gradient operand_gradient = 1; (Is this a per operand thing BTW?)
```

I'm wondering how many things would the current change break internally. But maybe that's fine. Perhaps other Googlers can comment? @thomasjoerg @ddunl

Reply:
As proposed the change is not backward compatible, you are right Tamas. I don't know what backward compatibility requirements this proto has TBH, but if in doubt, I'd prefer to keep compatibility the way you suggested.

Reply:
One more reason to separate the gradient from the precision is that we actually try to move away from using precision values and we may change over to using "algorithm" values soon. So it's possible that in the future, the operand_precisions will not be set. (In the first step we will still allow setting them as a backwards compatibility measure.)